### PR TITLE
Adds back native unit tests

### PR DESCRIPTION
**Pull request by ashfurrow**
This is based off [a discussion](https://github.com/artsy/emission/pull/1252#pullrequestreview-189045716) about adding native unit tests (and CI) _back_ to Emission. This PR is a WIP because the app won't even compile right now (competing subpecs for the `React` pod are causing issues) and the unit tests themselves weren't passing due to a MapBox error:

```
2019-01-04 17:31:38.744 Emission[33581:621372] [INFO] {com.mapbox.mbgl.DefaultFileSour}[Database]: cannot open file at line 38280 of [0e5ffd9123] (Code 14)
2019-01-04 17:31:38.745 Emission[33581:621372] [INFO] {com.mapbox.mbgl.DefaultFileSour}[Database]: [0e5ffd9123]:38280: (3) open(/Users/ash/Library/Developer/CoreSimulator/Devices/F6548399-EA8C-4AB5-B4A8-1FA0E4D1A0EE/data/Containers/Data/Application/06E86111-C4C5-414E-B906-E3201ADFE8E1/Library/Application Support/net.artsy.Emission/.mapbox/cache.db) -  (Code 14)
```

Once those two issues are resolved, we can add Circle CI to run native tests on every PR (I'll make it as fast as possible).

### CODE REVIEW

**Comment by ashfurrow** [comment]
Looks like I'm hitting this bug: https://github.com/CocoaPods/CocoaPods/issues/7195 We can't link with `RCTest` on our main app target because it won't launch on device (I think). I'm considering separating the unit test target from the Emission Example Xcode project to get around this.

**Comment by orta** [comment]
Do we need the test subspecies?

**Comment by ashfurrow** [comment]
`RCTest` subspec has the `RCTTestRunner` class that's needed by the existing tests.

**Comment by ashfurrow** [comment]
I'm going to keep working on speeding up the test suite for a little bit – running into issues with the Xcode project and Gemfile in a subdirectory. But this is ready for review.

**Comment by ashfurrow** [comment]
Sweet, got CI times down to 6.5 minutes.

File: package.json [hunk 1]
```diff
@@ -11,6 +11,7 @@
   "main": "index.ios.js",
   "scripts": {
     "bundle": "yarn generate-graphql-query-map && react-native bundle --platform=ios --dev=false --entry-file=index.ios.js --bundle-output Pod/Assets/Emission.js --sourcemap-output Pod/Assets/Emission.js.map --assets-dest Pod/Assets && rm -rf Pod/Assets/assets/node_modules",
+    "bundle:native-ci": "yarn generate-graphql-query-map && react-native bundle --platform=ios --dev=false --entry-file=index.tests.ios.js --bundle-output Pod/Assets/Emission.js --sourcemap-output Pod/Assets/Emission.js.map --assets-dest Pod/Assets && rm -rf Pod/Assets/assets/node_modules",
```
Comment (orta) [inline package.json]: I see what you're doing here, and prefer it, but better to be consistent with the rest of the file and use `-`s

**Pull request merged**